contracts/solidity/Oracles/FutarchyOracle.sol

```diff
@@ -128,12 +129,27 @@ contract FutarchyOracle is Oracle {
     {
         // Outcome is not set yet and deadline has passed
         require(!isSet && deadline <= now);
+
         uint[] memory outcomeTokenDistribution = getOutcomeTokenDistribution(markets[0]);
+
+        // require(
+        //     int(outcomeTokenDistribution[0]) >= 0 &&
+        //     int(outcomeTokenDistribution[1]) >= 0
+        // );
+        // int highest = int(outcomeTokenDistribution[0]).sub(int(outcomeTokenDistribution[1]));
         uint highest = outcomeTokenDistribution[0] - outcomeTokenDistribution[1];
+
         int highestIndex = 0;
         for (uint8 i=1; i<markets.length; i++) {
             outcomeTokenDistribution = getOutcomeTokenDistribution(markets[i]);
-            if ((outcomeTokenDistribution[0] - outcomeTokenDistribution[1]) > highest)
+
+            // require(
+            //     int(outcomeTokenDistribution[0]) >= 0 &&
+            //     int(outcomeTokenDistribution[1]) >= 0
+            // );
+
+            // if (int(outcomeTokenDistribution[0]).sub(int(outcomeTokenDistribution[1])) > highest)
+            if (outcomeTokenDistribution[0] - outcomeTokenDistribution[1] > highest)
                 highestIndex = i;
```

Comment:
@Georgi87 It seems like the test for this oracle actually _depends_ on the integer subtraction underflowing. This is probably a sign that something is wrong. In any case, I am unable to replace the subtraction with a safe version without the test failing due to a reversion (which happens when I `sub` with `uint`) or a different result (which happens when casting to `int` before `sub`ing).